include/deal.II/dofs/dof_accessor.templates.h

```diff
@@ -1588,6 +1588,223 @@ namespace internal
+
+        // 1) VERTEX dofs
+        for (const unsigned int vertex :
+             GeometryInfo<structdim>::vertex_indices())
+          for (unsigned int d = 0; d < dofs_per_vertex; ++d, ++index)
+            Operation::process_vertex_dof(
+              accessor, vertex, d, dof_indices[index], fe_index);
+
+        // 2) copy dof numbers from the LINE. for lines with the wrong
+        // orientation (which might occur in 3d), we have already made sure that
+        // we're ok by picking the correct vertices (this happens automatically
+        // in the vertex() function). however, if the line is in wrong
+        // orientation, we look at it in flipped orientation and we will have to
+        // adjust the shape function indices that we see to correspond to the
+        // correct (face/cell-local) ordering.
+        if (structdim == 2 || structdim == 3)
+          for (unsigned int line = 0;
+               line < GeometryInfo<structdim>::lines_per_cell;
+               ++line)
+            for (unsigned int d = 0; d < dofs_per_line; ++d, ++index)
+              Operation::process_dof(
+                *accessor.line(line),
+                accessor.get_fe(fe_index)
+                  .adjust_line_dof_index_for_line_orientation(
+                    d, accessor.line_orientation(line)),
+                dof_indices[index],
+                fe_index);
+
+        // 3) copy dof numbers from the FACE. for faces with the wrong
+        // orientation, we have already made sure that we're ok by picking the
+        // correct lines and vertices (this happens automatically in the line()
+        // and vertex() functions). however, if the face is in wrong
+        // orientation, we look at it in flipped orientation and we will have to
+        // adjust the shape function indices that we see to correspond to the
+        // correct (cell-local) ordering. The same applies, if the face_rotation
+        // or face_orientation is non-standard
+        if (structdim == 3)
+          for (unsigned int quad = 0;
+               quad < GeometryInfo<structdim>::quads_per_cell;
+               ++quad)
+            for (unsigned int d = 0; d < dofs_per_quad; ++d, ++index)
+              Operation::process_dof(
+                *accessor.quad(quad),
+                accessor.get_fe(fe_index)
+                  .adjust_quad_dof_index_for_face_orientation(
+                    d,
+                    accessor.face_orientation(quad),
+                    accessor.face_flip(quad),
+                    accessor.face_rotation(quad)),
+                dof_indices[index],
+                fe_index);
+
+        // 4) INNER dofs
+        for (unsigned int d = 0; d < inner_dofs; ++d, ++index)
+          Operation::process_dof(accessor, d, dof_indices[index], fe_index);
+
+        AssertDimension(dof_indices.size(), index);
+      }
+
+
+
+      template <typename DoFHandlerType, bool level_dof_access, int structdim>
+      class DoFIndexGetter
```

Comment:
Even though it is internal, I would still suggest to add a comment for this struct - and let it be a struct rather than a class because there is no access restriction.